### PR TITLE
[IMP] connector_dns: Code and test improvements

### PR DESCRIPTION
**Pull request by lasley**
@noahzaozao - Please don't merge this yet, I don't think it's actually usable quite yet. I just want you to see the changes, just in case I did something that goes a specific design decision. 

It would be nice if you could also attempt to hook this into your existing `dnspod` adapter before merge in order to make sure I didn't slice out anything essential. I think the main thing we may need to add back in are the buttons to trigger a manual check, but I ask why a cron to maintain an updated database will not work better?

Also, I'd like for all the commits in the main branch to be squashed before merging mine please. If we squash afterwards, it messes with my GPG signature.

Description of changes:

* Separate models by file
* Add `import_synchronizer`
* Add `export_synchronizer`
* Add `delete_synchronizer` (not complete)
* Add `mapper`
* Add `fail_date` to abstract bind, remove state
* Add `dns.record.type` model and default data
* Add external date field handling to binder
* Rename `dns.domain` to `dns.zone`
* Remove `line` field from `dns.record`
* Change inheritance structure, add a `{model_name}.bind` framework
* Add security
* Code cleanup
* Add test coverage
* Don't cover test assets
* Change bind field to `odoo_id` to ease future migration
* Improve ReadMe

cc @elicoidal @t3ddftw

OCA/infrastructure-dns#4

### CODE REVIEW

**Comment by lasley** [comment]
Hmmmm doesn't seem like we have coverage integration in this repo. Anyways, 91% at the moment - https://codecov.io/gh/laslabs/infrastructure-dns/branch/release%2F8.0%2Fconnector-dns

**Comment by elicoidal** [comment]
Probably we need to properly start the repo (travis.yml+coverall)

File: connector_dns/unit/export_synchronizer.py [hunk 1]
```diff
@@ -1,57 +1,374 @@
...
+_logger = logging.getLogger(__name__)
 
-Exporters for DNS.
+
+""" Exporters for Dns.
 
 In addition to its export job, an exporter has to:
 
-* check in DNS if the record has been updated more recently than the
+* check in Dns if the record has been updated more recently than the
```
Comment (lasley) [inline connector_dns/unit/export_synchronizer.py]: This was just hit by a sed - `DNS` is not PEP compliant, hence `Dns`. The docs strings should all say `DNS` though

File: connector_dns/tests/test_backend_adapter.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import mock
+import inspect
+
+from openerp.addons.connector_dns.unit import backend_adapter
+
+from .common import SetUpDnsBase
+
+
+_file = 'openerp.addons.connector_dns.unit.backend_adapter'
```
Comment (lasley) [inline connector_dns/tests/test_backend_adapter.py]: I wish, seems we're stuck with this in v10 as well. This is just the package path to the adapter

File: connector_dns/unit/export_synchronizer.py (hunk 1 above)
Comment (lasley) [inline connector_dns/unit/export_synchronizer.py]: Oh wait I had this backwards my bad. Wonder what language I was thinking of that uses lower case abbreviations in the CapWords convention. Regardless, changed

**Comment by lasley** [comment]
@elicoidal @noahzaozao - Alright so I've finished up with my AWS connector plans, and it seems these models will do fine from a preliminary level. I'd say we're good to call this frozen if you want to merge this & begin implementation on your end.

We are lacking a delete synchronizer, but I don't quite have a good one of those yet. I figure it's a non-issue at the moment, and doesn't block the progress of parallel development on the other connectors.

**Comment by elicoidal** [comment]
OK： @noahzaozao let's merge this one and ask for community review in the OCA repo

**Pull request merged**